lib/paperclip/attachment_extensions.rb

```diff
@@ -14,6 +14,15 @@ def process_style?(style_name, style_args)
       end
     end
 
+    def storage_schema_version
+      instance_read(:storage_schema_version) || 0
+    end
+
+    def assign_attributes
+      super
+      instance_write(:storage_schema_version, 1)
+    end
+
```

Comment:
I'm not super comfortable with that part, it looks a bit hackish to me and I'm not sure about the implications.
I guess it *should* work though.

Reply:
The alternative approach is adding a `before_post_process` hook to every model (or one of the concerns), but this is quite neat in that `instance_write` doesn't do anything if the model doesn't have the relevant column. Also using `super` is a bit more clean than copy-and-pasting the original method in full, but it's understandably a bit hackick because the whole thing is a monkey-patch.